src/app/snapp_test_transaction/snapp_test_transaction.ml

```diff
@@ -0,0 +1,1251 @@
+    ; amount = Currency.Amount.of_int 10000000000 (*10 Mina*)
+    }
+  in
+  let%bind parties =
+    Transaction_snark.For_tests.create_trivial_predicate_snapp
+      ~constraint_constants spec ledger
+  in
+  Core.printf "Snapp transaction yojson: %s\n\n%!"
+    (Parties.to_yojson parties |> Yojson.Safe.to_string) ;
+  Core.printf "Snapp transaction graphQL input %s\n\n%!"
+    (graphql_snapp_command parties) ;
+  Core.printf "Updated accounts\n" ;
+  let consensus_constants =
+    Consensus.Constants.create ~constraint_constants
+      ~protocol_constants:Genesis_constants.compiled.protocol
+  in
+  let state_body =
+    let compile_time_genesis =
+      (*not using Precomputed_values.for_unit_test because of dependency cycle*)
+      Mina_state.Genesis_protocol_state.t
+        ~genesis_ledger:Genesis_ledger.(Packed.t for_unit_tests)
+        ~genesis_epoch_data:Consensus.Genesis_epoch_data.for_unit_tests
+        ~constraint_constants ~consensus_constants
+    in
+    compile_time_genesis.data |> Mina_state.Protocol_state.body
+  in
+  let witnesses =
+    Transaction_snark.parties_witnesses_exn ~constraint_constants ~state_body
+      ~fee_excess:Currency.Amount.Signed.zero
+      ~pending_coinbase_init_stack:Pending_coinbase.Stack.empty (`Ledger ledger)
+      [ parties ]
+  in
+  let open Async.Deferred.Let_syntax in
+  let module T = Transaction_snark.Make (struct
+    let constraint_constants = constraint_constants
+
+    let proof_level = proof_level
+  end) in
+  let%map _ =
+    Async.Deferred.List.fold ~init:((), ()) (List.rev witnesses)
+      ~f:(fun _ ((witness, spec, statement, snapp_statement) as w) ->
+        Core.printf "%s"
+          (sprintf
+             !"current witness \
+               %{sexp:(Transaction_witness.Parties_segment_witness.t * \
+               Transaction_snark.Parties_segment.Basic.t * \
+               Transaction_snark.Statement.With_sok.t * (int * \
+               Snapp_statement.t)option) }%!"
+             w) ;
+        let%map _ =
+          T.of_parties_segment_exn ~snapp_statement ~statement ~witness ~spec
+        in
+        ((), ()))
+  in
+  ()
+
+module Flags = struct
+  open Command
+
+  let default_fee = Currency.Fee.of_formatted_string "1"
+
+  let min_fee = Currency.Fee.of_formatted_string "0.003"
+
+  let memo =
+    Param.flag "--memo" ~doc:"STRING Memo accompanying the transaction"
+      Param.(optional string)
+
+  let fee =
+    Param.flag "--fee"
+      ~doc:
+        (Printf.sprintf
+           "FEE Amount you are willing to pay to process the transaction \
+            (default: %s) (minimum: %s)"
+           (Currency.Fee.to_formatted_string default_fee)
+           (Currency.Fee.to_formatted_string min_fee))
+      (Param.optional txn_fee)
+
+  let snapp_account_key =
+    Param.flag "--snapp-account-key" ~doc:"Public Key of the new snapp account"
+      Param.(required public_key_compressed)
+
+  let amount =
+    Param.flag "--receiver-amount" ~doc:"Receiver amount"
```

Comment:
doc should be something like:
```
NN Receiver amount in Mina
```
(maybe it's nanoMina)

As it is now, the help prints out
``
  --receiver-amount Receiver   amount
```